ssl/ssl_lib.c

```diff
@@ -3025,12 +3038,30 @@ SSL_CTX *SSL_CTX_new(const SSL_METHOD *meth)
 
     return ret;
  err:
-    SSLerr(SSL_F_SSL_CTX_NEW, ERR_R_MALLOC_FAILURE);
+    SSLerr(SSL_F_SSL_CTX_NEW_INT, ERR_R_MALLOC_FAILURE);
  err2:
     SSL_CTX_free(ret);
     return NULL;
 }
 
+#undef SSL_CTX_new
+/* Exported for the sake of programs linked with pre-1.1.1 OpenSSL */
+SSL_CTX *SSL_CTX_new(const SSL_METHOD *meth)
+{
+    return ssl_ctx_new_int(meth, TLS1_2_VERSION);
+}
+
+/* Exported for programs linked with OpenSSL 1.1.1 */
+SSL_CTX *SSL_CTX_new_111(const SSL_METHOD *meth)
```

Comment:
I can't say I'm massively fond of this name. What happens when we have 1.1.2 or 1.1.3...or...

Maybe SSL_CTX_new_ex()?

Reply:
SSL_CTX_ new_tls13()?

Reply:
why not SSL_CTX_new_ex?
it is not yet specified which paramters it will have.

Reply:
Exactly because of that, and because I see this as an interim solution that should disappear with 1.2.0 at the latest.